snowplow-tracker/src/main/java/com/snowplowanalytics/core/ecommerce/EcommerceEvent.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.core.ecommerce
+
+import com.snowplowanalytics.core.constants.Parameters
+import com.snowplowanalytics.core.constants.TrackerConstants
+import com.snowplowanalytics.snowplow.ecommerce.entities.Product
+import com.snowplowanalytics.snowplow.ecommerce.entities.Promotion
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+
+/**
+ * Interface for creating Ecommerce entities.
+ */
+interface EcommerceEvent {
+    fun productToSdj(product: Product) : SelfDescribingJson {
```

Comment:
Suggestion: maybe the serialization could be implemented in the Product class? I think that would  better encapsulate it. As an example, [here is how I did similar serialization in the media APIs](https://github.com/snowplow/snowplow-android-tracker/blob/issue/606-media/snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/media/entity/MediaPlayerEntity.kt#L64-L84).